src/python_framework_v02/troute/nhd_io.py

```diff
@@ -233,6 +238,99 @@ def get_ql_from_wrf_hydro(qlat_files, index_col="station_id", value_col="q_later
 
     return ql
 
+def read_netcdfs(files, dim, transform_func=None):
+    def process_one_path(path):
+        with xr.open_dataset(path) as ds:
+            if transform_func is not None:
+                ds = transform_func(ds)
+            ds.load()
+            return ds
+
+    paths = sorted(glob.glob(files))
+    datasets = [process_one_path(p) for p in paths]
+    combined = xr.concat(datasets, dim)
+    return combined
+
+def add_time_dimension(ds):
+    t = [ds['queryTime'].values[0]]
+    time_da = xr.DataArray(t,[('time',t)])
+    return ds.drop_vars('time').expand_dims(time=time_da,axis=1)
+
+def get_usgs_from_wrf_hydro(routelink_subset_file,usgs_timeslices_folder):
+    usgs_file_pattern_filter = "2020-03-21*.usgsTimeSlice.ncdf"
```

Comment:
pull the file pattern filter from the `data_assimilation_parameters` dictionary.